task/slack-webhook-notification/0.1/slack-webhook-notification.yaml

```diff
@@ -41,12 +49,140 @@ spec:
           value: $(params.key-name)
         - name: MESSAGE
           value: $(params.message)
+      args:
+        - --files
+        - $(params.files[*])
+        - --submodules
+        - $(params.submodules[*])
       script: |
         #!/usr/bin/env bash
+
+        # ---------
+        #  HELPERS
+        # ---------
+
+        function escapeDoubleQuote { echo "${1//\"/\\\"}"; }
+
+
+        # ---------
+        #  DUMPERS
+        # ---------
+
+        function dumpSeparator {
+          echo "-----------------------------------------"
+        }
+
+        function dumpFile {
+          filePath=$1
+
+          read -r -d '' ret << EOM
+        *${filePath}:*
+        \`\`\`
+        $(cat "${filePath}")
+        \`\`\`
+        EOM
+
+          echo "${ret}"
+        }
+
+        function dumpSubmodule {
+          name=$1
+
+          path=$(git config -f .gitmodules --get submodule."${name}".path)
+          url=$(git config -f .gitmodules --get submodule."${name}".url)
+
+          current_commit=$(git -C "${path}" rev-parse HEAD)
+          previous_commit=$(git diff HEAD~1 "${path}" | grep commit | head -n 1 | awk '{print $3;}')
```

Comment:
TBH I don't know to detect efficiently if the submodule was updated. So I assume if it was, the merge commit will got it and then, in this case, 1 commit is enough. It's not perfect but I don't see an easy way to get this information

Reply:
Hmm, so it will work correctly if
* this runs in a push pipeline and the repo uses merge commits _or_
* updates to submodules are done in separate PRs, all submodules are updated in the same commit

That's probably fine, let's just document the shortcomings in the task description

Reply:
README updated